### PR TITLE
forms tests

### PR DESCRIPTION
**Pull request by zouxuoz**

### CODE REVIEW

File: client/containers/ui/Form/__tests__/Field.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  const mountWithFormContext = (component, customFormContext) => mount(component, {
+    context: { form: { ...formContext, ...customFormContext } },
+  });
+
+  it('should call form change after mount', () => {
+    const onFormChange = jest.fn();
+
+    mountWithFormContext(<Field name="foo" component={() => (<input />)} />, {
+      fields: {
+        foo: [],
```
Comment (radiansz) [inline client/containers/ui/Form/__tests__/Field.js]: Why array?

File: client/containers/ui/Form/__tests__/Field.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        values: { foo: value },
+      });
+
+      inputText.find('component').prop('input').onChange({ target: { value: 'qux' } });
+      inputText.find('component').prop('input').onChange({ target: { value } });
+
+      expect(inputText.find('component').prop('meta').dirty).toEqual(false);
+    });
+
+    it('should untoched at initial state', () => {
```
Comment (radiansz) [inline client/containers/ui/Form/__tests__/Field.js]: Misprint -- touch

File: client/containers/ui/Form/__tests__/Field.js (hunk 1 above)
Comment (zouxuoz) [inline client/containers/ui/Form/__tests__/Field.js]: fixed

File: client/containers/ui/Form/__tests__/Field.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+    const value = 'bar';
+
+    inputText.find('component').prop('input').onChange({ target: { value } });
+
+    expect(inputText.find('component').prop('input').value).toEqual(value);
+  });
+
+  describe('meta', () => {
+    it('should clean at initial state', () => {
```
Comment (zouxuoz) [inline client/containers/ui/Form/__tests__/Field.js]: fixed

File: client/containers/ui/Form/__tests__/Field.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+      expect(inputText.find('component').prop('meta').dirty).toEqual(false);
+    });
+
+    it('should dirty after change component value', () => {
+      const inputText = mountWithFormContext(<Field name="foo" component={() => (<input />)} />, {
+        fields: { foo: [] },
+      });
+
+      const value = 'bar';
```
Comment (zouxuoz) [inline client/containers/ui/Form/__tests__/Field.js]: fixed

File: client/containers/ui/Form/__tests__/Field.js (hunk 2 above)
Comment (zouxuoz) [inline client/containers/ui/Form/__tests__/Field.js]: fixed

**Pull request closed without merging**